vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+
+The `enw_fit_opts` allows you to set several critical components for the Stan computational engine.
+These parameters are passed to Stan during the model fitting process and have an influence on computation time and quality of the model fit.
+Knowing when and if the defaults need to be changed is an important part of the Bayesian workflow.[@gelmanBayesianWorkflow2020]
+All of the parameters are passed to the `sampling` function available from [`{cmdstanr}`](https://mc-stan.org/cmdstanr/reference/model-method-sample.html).
+
+# Failures modes
+
+In this section we will discuss the most common issues when working with `{epinowcast}` specifically and CmdStan/ Bayesian inference more generally.
+The solutions to these issues typically require you to investigate your [sampler settings](#samplerParams), [model settings](#modelParams), or [interogate your data](#modelData) further.
+
+## Sampler settings {.samplerParams}
+
+`{epinowcast}` provides users the ability to pass arguments directly to CmdStan which can improve model run time and produce better inferences.
+These parameters can be passed the `enw_fit_opts` function.
+
+### `threads_per_chain` and `chains`
+
+One of the key components of Bayesian inference is creating draws from independent Markov chains.
+To assess convergence and the reliability of a given posterior distribution it is vital that at least two chains are used with **`4` chains** being the more common default.
+
+Stan allows for multi-threading and the Stan code used within `{epinowcast}` has tried to take advantage of these capabilities where possible.
+Multi-theading allows for the calculations to be spread across multiple threads, which could accelerate inference by allowing embarassingly parallel calculations to be spread across threads and then later combined.
+If possible, a good default is `2`, however, if you have available compute on your machine, you can increase this to a higher value and may see some fit time reductions.
+Please note that the `theads_per_chain` has not impact on model convergence or fittings and is purely a means of speeding up the model fitting time.
+
+::: {.alert .alert-warning}
+**The product of the `thread_per_chain` and `chains`** arguments should be equal to or less than the number of cores on your machine! 
+:::
+
+### iter_warmup and iter_sampling {.iters}
+
+The number of samples using for the warmup phase of Stan and the number of sampling iterations are controlled by the `iter_warmup` and `iter_sampling` arguments, respectively.
+Stan requires a sufficient number of warmup iterations to dynamically estimate the step change sizes during model fitting. 
+Unlike with Gibbs sampling (such as the method used in JAGS), you do not need to set large values for `iter_warmup` or `iter_sampling`.
+Conversely, setting large values for iterations is a sign of poor model fit and indicates that other approaches need to be employed (e.g., `priors`, `adapt_delta`, `max_treedepth`)
```

Comment:
I don't know that this is necessarily true. ESS / iter < 0.1 is certainly a red flag. But, if you're fitting a challenging model and interested in the tails and getting an ESS_tail / iter \approx 0.25, I don't think it's unreasonable to use ~5000 pre-warmup samples ~5000 post-warmup samples (say 5 chains, 1000 per chain).

Reply:
Some stuff from stan convergence diagnostics: https://mc-stan.org/rstan/reference/Rhat.html#:~:text=Both%20bulk%2DESS%20and%20tail,respective%20posterior%20quantiles%20are%20reliable.

Reply:
Sure, but if you're interested in the extremes of the tail quantiles and want reliable estimates, you're interested in the MCSE of the expectation of the quantile, no? Even if you have convergence, an $n_{eff}$ of only 100 should reflect some sampler noise.